a10_octavia/controller/worker/tasks/cert_tasks.py

```diff
@@ -63,143 +64,153 @@ class SSLKeyCreate(task.Task):
 
 
 class SSLCertDelete(task.Task):
     """Task to delete an SSL certificate"""
     @axapi_client_decorator
     def execute(self, loadbalancer, listener, vthunder):
-        barbican_client = BarbicanACLAuth().get_barbican_client(loadbalancer.project_id)
-        cert_data = utils.get_cert_data(barbican_client, listener)
-        try:
-            if self.axapi_client.file.ssl_cert.exists(file=cert_data.cert_filename):
-                self.axapi_client.file.ssl_cert.delete(file=cert_data.cert_filename)
-                LOG.debug("SSL certificate deleted successfully: %s", cert_data.cert_filename)
-        except Exception as e:
-            LOG.warning("Failed to delete SSL Certificate: %s", str(e))
-        return cert_data
+        if listener.protocol == "TERMINATED_HTTPS":
+            barbican_client = BarbicanACLAuth().get_barbican_client(loadbalancer.project_id)
+            cert_data = utils.get_cert_data(barbican_client, listener)
+            try:
+                if self.axapi_client.file.ssl_cert.exists(file=cert_data.cert_filename):
+                    self.axapi_client.file.ssl_cert.delete(file=cert_data.cert_filename)
+                    LOG.debug("SSL certificate deleted successfully: %s", cert_data.cert_filename)
+            except Exception as e:
+                LOG.warning("Failed to delete SSL Certificate: %s", str(e))
+            return cert_data
 
 
 class SSLKeyDelete(task.Task):
     """Task to delete an SSL Key"""
     @axapi_client_decorator
     def execute(self, cert_data, vthunder):
-        try:
-            if self.axapi_client.file.ssl_key.exists(file=cert_data.key_filename):
-                self.axapi_client.file.ssl_key.delete(file=cert_data.key_filename)
-                LOG.debug("SSL Key deleted successfully: %s", cert_data.key_filename)
-        except Exception as e:
-            LOG.warning("Failed to delete SSL Key: %s", str(e))
+        if cert_data is not None:
+            try:
+                if self.axapi_client.file.ssl_key.exists(file=cert_data.key_filename):
+                    self.axapi_client.file.ssl_key.delete(file=cert_data.key_filename)
+                    LOG.debug("SSL Key deleted successfully: %s", cert_data.key_filename)
+            except Exception as e:
+                LOG.warning("Failed to delete SSL Key: %s", str(e))
 
 
 class ClientSSLTemplateDelete(task.Task):
     """Task to delete a client ssl template for a listener"""
     @axapi_client_decorator
     def execute(self, cert_data, vthunder):
-        try:
-            if self.axapi_client.slb.template.client_ssl.exists(name=cert_data.template_name):
-                self.axapi_client.slb.template.client_ssl.delete(name=cert_data.template_name)
-                LOG.debug("Client SSL Template deleted successfully: %s",
-                          cert_data.template_name)
-        except Exception as e:
-            LOG.warning("Failed to delete Client SSL Template: %s", str(e))
+        if cert_data is not None:
```

Comment:
This shouldn't be needed. We've designed the flows so that cert_data will never be none if this task is executed.